bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/SalusApi.java

```diff
@@ -0,0 +1,224 @@
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.salus.internal.rest;
+
+import static java.util.Objects.requireNonNull;
+import static org.openhab.binding.salus.internal.rest.ApiResponse.error;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The SalusApi class is responsible for interacting with a REST API to perform various operations related to the Salus
+ * system. It handles authentication, token management, and provides methods to retrieve and manipulate device
+ * information and properties.
+ *
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class SalusApi {
+    private static final int MAX_TIMES = 3;
+    private final Logger logger;
+    private final String username;
+    private final char[] password;
+    private final String baseUrl;
+    private final RestClient restClient;
+    private final GsonMapper mapper;
+    @Nullable
+    private AuthToken authToken;
+    @Nullable
+    private LocalDateTime authTokenExpireTime;
+    private final Clock clock;
+
+    public SalusApi(String username, char[] password, String baseUrl, RestClient restClient, GsonMapper mapper,
+            Clock clock) {
+        this.username = requireNonNull(username, "username");
+        this.password = requireNonNull(password, "password");
+        this.baseUrl = removeTrailingSlash(requireNonNull(baseUrl, "baseUrl"));
+        this.restClient = requireNonNull(restClient, "restClient can not be null!");
+        this.mapper = requireNonNull(mapper, "mapper can not be null!");
+        this.clock = requireNonNull(clock, "clock can not be null!");
+        // thanks to this, logger will always inform for which rest client it's doing the job
+        // it's helpful when more than one SalusApi exists
+        logger = LoggerFactory.getLogger(SalusApi.class.getName() + "[" + username.replace(".", "_") + "]");
+    }
+
+    public SalusApi(String username, char[] password, String baseUrl, RestClient restClient, GsonMapper mapper) {
+        this(username, password, baseUrl, restClient, mapper, Clock.systemDefaultZone());
+    }
+
+    private RestClient.Response<@Nullable String> get(String url, RestClient.Header header, int times) {
+        refreshAccessToken();
+        var response = restClient.get(url, authHeader());
+        if (response.statusCode() == 401) {
+            logger.info("Refreshing access token");
+            login(username, password);
+            if (times > MAX_TIMES) {
+                logger.warn("Could not refresh access token after {} times", MAX_TIMES);
```

Comment:
```suggestion
                logger.debug("Could not refresh access token after {} times", MAX_TIMES);
```
Should be debug as this situation leads to an unsuccesfull api call and it will set the bridge to offline + detail. (hopefully)